pymeasure/instruments/wentworthlabs/s200.py

```diff
@@ -0,0 +1,548 @@
+
+    def __init__(self,
+                 adapter,
+                 name="Wentworth Labs S200 Probe Table",
+                 write_termination="\n",
+                 read_termination="\n",
+                 query_delay=0.5,
+                 write_delay=0.5,
+                 timeout=5000,
+                 **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            write_termination=write_termination,
+            read_termination=read_termination,
+            send_end=True,
+            includeSCPI=True,
+            timeout=timeout,
+            **kwargs
+        )
+        self.write_delay = write_delay
+        self.query_delay = query_delay
+        self.last_write_timestamp = 0.0
+        self.command_execution_info = S200.ExecutionInfoCode(0)
+
+    chuck_lift = Instrument.setting(
+        "%s",
+        "Control the chuck to the fine down (CDW) or fine up (CUP) position.",
+        # The GDW/GUP command is used to move the chuck to the Gross Up or Gross
+        # Down positions.
+
+        validator=strict_discrete_set,
+        values={True: 'CUP', False: 'CDW'},
+        map_values=True,
+        check_set_errors=True
+    )
+
+    chuck_gross_lift = Instrument.setting(
+        "%s",
+        "Control the chuck to the gross down (CDW) or fine up (CUP) position.",
+        # The CDW/CUP command is used to move the chuck to the Fine Down or
+        # Fine Up positions. The GUP command may be used to move the chuck to a
+        # safe gross lift height ready for probing the first die. Subsequent Z movements
+        # to move the probes onto and off of the die under test should use the CUP and
+        # CDW command. If edge sensors are being used this will dynamically adjust
+        # the gross lift value to track the wafer surface. Therefore it is advisable not to
+        # use the GUP command during the probing process until the start of the next
+        # wafer. Doing so may cause unexpected results.
+        # AWP compatible: Yes
+        validator=strict_discrete_set,
+        values={True: 'GUP', False: 'GDW'},
+        map_values=True,
+        check_set_errors=True
+    )
+
+    chuck_override = Instrument.setting(
+        "%s",
+        "Control the chuck override of the main chuck of the probe station.",
+        validator=strict_discrete_set,
+        values={True: 'CO1', False: 'CO0'},
+        map_values=True,
+        check_set_errors=True
+    )
+
+    lamp_on = Instrument.setting(
+        "%s",
+        "Control lamp on/off of the probe station",
+        validator=strict_discrete_set,
+        values={True: 'LI1', False: 'LI0'},
+        map_values=True,
+        check_set_errors=True
+    )
+
+    x_position = Instrument.control(
+        "PSS X", "GTS X,%d",
+        "Control the x-axis position in microns",
+        validator=strict_range,
+        values=X_POS_VALID_RANGE,
+        get_process=lambda r:
+        float(r[1]),
```

Comment:
It seems the lambdas are auto-formatted in multi line when they  should be on a single line.

What is `r[0]` in the reply message? It may be that a uniform lambda passed to `preprocess_reply` argument would be useful for all these commands that ignore the first element from the response.

For instance, `x_position` and `xy_position` can have the same `preprocess_reply` instead of `xy_position` having a custom `get_process`.

Reply:
Just checked out the tests and I see why you ignore the first element.
Yea, I think a `preprocess_reply` may be a better approach than `get_process`.
```
    x_position = Instrument.control(
        "PSS X", "GTS X,%d",
        "Control the x-axis position in microns",
        validator=strict_range,
        values=X_POS_VALID_RANGE,
        preprocess_reply=lambda r: r[r.index(',')+1:],
```

That should `preprocess_reply` would work for all the position properties. Before parsing the reply string for values, ignore characters up to the first comma. It will return single floats or list of floats. You can do something similar for other properties:

```
    y_index = Instrument.control(
        "RYM", "WYM %d",
        "Control the y index of the chuck in units of 10 microns ",
        # AWP compatible: Yes
        validator=strict_range,
        values=X_Y_INDEX_VALID_RANGE,
        check_set_errors=True,
        preprocess_reply=lambda r: r.replace("RYM ", "")
    )
```

Reply:
many thanks